src/CoreWCF.Primitives/src/CoreWCF/Channels/TransportSecurityBindingElement.cs

```diff
@@ -0,0 +1,216 @@
+                    this.EndpointSupportingTokenParameters.Endorsing.RemoveAt(0);
+                    try
+                    {
+                        base.ConfigureProtocolFactory(sessionFactory, credentialsManager, true, issuerBindingContext, context.Binding);
+                    }
+                    finally
+                    {
+                        this.EndpointSupportingTokenParameters.Endorsing.Insert(0, scParameters);
+                    }
+
+                    securityServiceDispatcher.SessionMode = true;
+                    securityServiceDispatcher.SessionServerSettings.InactivityTimeout = this.LocalServiceSettings.InactivityTimeout;
+                    securityServiceDispatcher.SessionServerSettings.KeyRolloverInterval = this.LocalServiceSettings.SessionKeyRolloverInterval;
+                    securityServiceDispatcher.SessionServerSettings.MaximumPendingSessions = this.LocalServiceSettings.MaxPendingSessions;
+                    securityServiceDispatcher.SessionServerSettings.MaximumKeyRenewalInterval = this.LocalServiceSettings.SessionKeyRenewalInterval;
+                    securityServiceDispatcher.SessionServerSettings.TolerateTransportFailures = this.LocalServiceSettings.ReconnectTransportOnFailure;
+                    securityServiceDispatcher.SessionServerSettings.CanRenewSession = scParameters.CanRenewSession;
+                    securityServiceDispatcher.SessionServerSettings.IssuedSecurityTokenParameters = scParameters.Clone();
+                    ((SecureConversationSecurityTokenParameters)securityServiceDispatcher.SessionServerSettings.IssuedSecurityTokenParameters).IssuerBindingContext = issuerBindingContext;
+                    securityServiceDispatcher.SessionServerSettings.SecurityStandardsManager = sessionFactory.StandardsManager;
+                    securityServiceDispatcher.SessionServerSettings.SessionProtocolFactory = sessionFactory;
+                    securityServiceDispatcher.SecurityProtocolFactory = (SecurityProtocolFactory)sessionFactory;
+
+                    // pass in the error handler for handling unknown security sessions - dont do this if the underlying channel is duplex since sending 
+                    // back faults in response to badly secured requests over duplex can result in DoS.
+                    //if (context.BindingParameters != null && context.BindingParameters.Find<IChannelDemuxFailureHandler>() == null
+                    //    && !IsUnderlyingListenerDuplex<TChannel>(context))
+                    //{
+                    //    context.BindingParameters.Add(new SecuritySessionServerSettings.SecuritySessionDemuxFailureHandler(sessionFactory.StandardsManager));
+                    //}
+                }
+                else
+                {
+                    //TODO later 
+                    //TransportSecurityProtocolFactory protocolFactory = new TransportSecurityProtocolFactory();
```

Comment:
This is an important scenario.

Reply:
incorporated... let's add a test.. will sync